source/UberRides/TokenManager.swift

```diff
@@ -143,36 +150,38 @@ import Foundation
         deleteCookies()
         let success = keychainWrapper.deleteObjectForKey(tokenIdentifier)
         if success {
+            NSUserDefaults.standardUserDefaults().removeObjectForKey(kCheckFirstRun)
```

Comment:
If we delete the first run key whenever you delete a key, the next key you fetch is going to be nil.

For example, we could save 2 keys with 2 separate identifiers and then delete one of them. When we try to access the second key, the `fetch()` check will think it's a first run and delete the key